include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3123,6 +3123,248 @@ namespace internal
 
 
 
+      template <int dim, int spacedim, bool level_dof_access>
+      static bool
+      p_refine_flag_set(const DoFCellAccessor<DoFHandler<dim, spacedim>,
+                                              level_dof_access> &accessor)
+      {
+        (void)accessor;
+        // ::DoFHandler does not support p-refinement
+        Assert(
+          false,
+          ExcMessage(
+            "DoFHandler is not of type hp::DoFHandler and does not support p-refinement."));
```

Comment:
You seem to use the same exception message quite a few times. Maybe it warrants being made into its own exception?